magda-typescript-common/src/OpaCompileResponseParser.ts

```diff
@@ -968,33 +970,68 @@ export default class OpaCompileResponseParser {
             return [];
         }
         this.data = this.data.result;
-        if (!_.isArray(this.data.support) || !this.data.support.length) {
+        if (
+            (!this.data.queries ||
+                !_.isArray(this.data.queries) ||
+                !this.data.queries.length) &&
+            (!_.isArray(this.data.support) || !this.data.support.length)
+        ) {
             // --- mean no rule matched
             return [];
         }
+
+        const queries = this.data.queries;
+
+        if (queries) {
+            queries.forEach((q: any, i: number) => {
```

Comment:
Probably easier (and more generic as we can handle any response no matter whether `support` / `queries` are there or not) to convert `this.data.queries` (answer) to a package.
Compared with packages in `support`, the answer/queries section contains rules body without package namespace.
We probably can put all answer queries into a fake namespace `magda.query.answer` (package path `magda.query` & fake rule name `answer`).
And add it to the support packages (if support is empty, we just create an array with package name `magda.query.answer`) so the same rule drying logic can be applied.
Then, we can call `evaluateRule` the fake rule `magda.query.answer`

Ignore me if the current code works well already 😄

Reply:
I wouldn't say well, but it does work afaik.

Maybe write an issue laying out a design for what you mean? I don't think it makes sense to invest any more time into auth for the content api right now, It'd be cool to have for later :).

Reply:
👍